### PR TITLE
[luci/pass] Enhance RemoveUnnecessaryTransposeNetPass test verification

### PR DESCRIPTION
**Pull request by zetwhite**
This PR is to enhance RemoveUnnecessaryTransposeNetPass tests-verification by checking transpose node is removed as expected.

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com>

* motivation : https://github.com/Samsung/ONE/pull/11891#discussion_r1386125771 
* issue ( but not highly related ) :  https://github.com/Samsung/ONE/issues/11654

### CODE REVIEW

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp [hunk 1]
```diff
@@ -262,4 +281,5 @@ TEST(RemoveUnnecessaryTransposeNetPass, basic_pattern3_NEG)
                         /*out*/ {1, 2, 3, 4});
 
   EXPECT_FALSE(pass.run(g.g()));
+  EXPECT_FALSE(is_transpose_removed(g.g()));
```
Comment (zetwhite) [inline compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp:284]: It still passes tests as expected. 

```
➜ ./build/compiler/luci/pass/luci_pass_test --gtest_filter='RemoveUnnecessaryTransposeNetPass.*'   
Running main() from /home/sseung/Workspace/ONE_second/externals/GTEST/googletest/src/gtest_main.cc
Note: Google Test filter = RemoveUnnecessaryTransposeNetPass.*
[==========] Running 6 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 6 tests from RemoveUnnecessaryTransposeNetPass
[ RUN      ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern1
[       OK ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern1 (1 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern2
[       OK ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern2 (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.identity_pattern
[       OK ] RemoveUnnecessaryTransposeNetPass.identity_pattern (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.basic_pattern1_NEG
[       OK ] RemoveUnnecessaryTransposeNetPass.basic_pattern1_NEG (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.basic_pattern2_NEG
[       OK ] RemoveUnnecessaryTransposeNetPass.basic_pattern2_NEG (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.basic_pattern3_NEG
[       OK ] RemoveUnnecessaryTransposeNetPass.basic_pattern3_NEG (0 ms)
[----------] 6 tests from RemoveUnnecessaryTransposeNetPass (1 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 6 tests.
```

**Review by seanshpark — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp [hunk 2]
```diff
@@ -136,6 +136,20 @@ class TransposeReshapeTransposeGraph : public TestIOGraph
   luci::CircleConst *_back_perm = nullptr;
 };
 
+bool is_transpose_removed(loco::Graph *g)
+{
+  bool is_transpose_removed = true;
```
Comment (ragmani) [inline compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp:141]: This name is duplicated with function name. How about changing the name to `ret`?

Comment (seanshpark) [reply compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp:141]: this function searches Transpose Op.
maybe we can use "transpose_exist" and return "not transpose_exist"

Comment (ragmani) [reply compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp]: Anyway, it would be better to avoid duplication of names.

**Review by ragmani — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp (hunk 2 above)
Comment (zetwhite) [reply compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp]: Thanks for pointing out. 
updated here - https://github.com/Samsung/ONE/pull/12077/commits/51002fe1eedcf578971e30bcca9603ca966adf35

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**